docs/PORTABILITY_AND_STABILITY.md

```diff
@@ -15,7 +15,11 @@ The stable interfaces are:
 
 * **The command line interface** of `systemd`, `systemctl`, `loginctl`, `journalctl`, and all other command line utilities installed in `$PATH` and documented in a man page. We will make sure that scripts invoking these commands will continue to work with future versions of systemd. Note however that the output generated by these commands is generally not included in the promise, unless it is documented in the man page. Example: the output of `systemctl status` is not stable, but that of `systemctl show` is, because the former is intended to be human readable and the latter computer readable, and this is documented in the man page.
 
-* **The protocol spoken on the socket referred to by `$NOTIFY_SOCKET`**, as documented in [sd_notify(3)](https://www.freedesktop.org/software/systemd/man/sd_notify.html).
+* **The protocol spoken on the socket referred to by `$NOTIFY_SOCKET`**, as documented in
+  [sd_notify(3)](https://www.freedesktop.org/software/systemd/man/sd_notify.html). Note that, although using
+  libsystemd should be preferred, this protocol can also be trivially reimplemented without external
```

Comment:
i would tone this down. I am not sure it should be "preferred". It's a good choice, so much i would say. But I think it's really not a clearcut situation if it should be "preferred" or not. If you only want sd_notify() and nothing else of libsystemd then I'd always suggest people to reimplement it on their own. If the use for example sd-bus too, then I think using our sd_notify() is a good idea too. 

Things like glib or Apache's APR and similar framework too should always implement the stuff on their own.

Then, non-C projects should also do their own implementatin instead of wrapping libsystemd just for this.

Hence, I'd tone this down and say "good choice", and that's it. A complex discussion when to do what sounds like a bit too much for these docs.

Reply:
Agreed. And since you're linking to a 167 line file, I think "trivially" should be struck too.

Reply:
I toned it down, but to avoid code duplication we should gently nudge toward using the library, especially when it's already used for other APIs.